homeassistant/components/fully_kiosk/sensor.py

```diff
@@ -0,0 +1,115 @@
+"""Fully Kiosk Browser sensor."""
+from __future__ import annotations
+
+from typing import Any
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import DATA_MEGABYTES, PERCENTAGE
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity import EntityCategory
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from .const import DOMAIN
+from .coordinator import FullyKioskDataUpdateCoordinator
+from .entity import FullyKioskEntity
+
+SENSORS: tuple[SensorEntityDescription, ...] = (
+    SensorEntityDescription(
+        key="batteryLevel",
+        name="Battery Level",
+        device_class=SensorDeviceClass.BATTERY,
+        native_unit_of_measurement=PERCENTAGE,
+    ),
+    SensorEntityDescription(key="screenOrientation", name="Screen Orientation"),
+    SensorEntityDescription(
+        key="foregroundApp",
+        name="Foreground App",
+    ),
+    SensorEntityDescription(key="currentPage", name="Current Page"),
+    SensorEntityDescription(
+        key="wifiSignalLevel",
+        name="WiFi Signal Level",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        device_class=SensorDeviceClass.SIGNAL_STRENGTH,
+    ),
+    SensorEntityDescription(
+        key="internalStorageFreeSpace",
+        name="Internal Storage Free Space",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+    SensorEntityDescription(
+        key="internalStorageTotalSpace",
+        name="Internal Storage Total Space",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+    SensorEntityDescription(
+        key="ramFreeMemory",
+        name="RAM Free Memory",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+    SensorEntityDescription(
+        key="ramTotalMemory",
+        name="RAM Total Memory",
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=DATA_MEGABYTES,
+    ),
+)
+
+STORAGE_SENSORS = [
+    "internalStorageFreeSpace",
+    "internalStorageTotalSpace",
+    "ramFreeMemory",
+    "ramTotalMemory",
+]
```

Comment:
So this is actually something we made entity descriptions for. They are extendable :)

So you could add, e.g., lambda or reference a function to these descriptions that do that calculation.